Katello/katello
# Fixes #34398 - exclude source redhat containers by default

#### What are the changes introduced in this pull request?
- Adds the ability to both exclude and include container tags when syncing.
- The `exclude_tags` are pre-populated with `*-source` to exclude the large source tags by default.
- `docker_tags_whitelist` is deprecated.

#### Considerations taken when implementing this change?
`docker_tags_whitelist` should still work in the API for now until we completely remove it.

#### What are the testing steps for this pull request?
1) Create a repo with no include or exclude tags via Hammer and the UI.  Note that the exclude tags get pre-populated with `*-source`.
2) Create a repo with include and exclude tags via Hammer and the UI.
3) Update a repo's include and exclude tags the Hammer and the UI.
4) Ensure that syncing still works properly.
5) Check the Pulp remote to see the updated include and exclude tags.
